dynamic/clientset/dynamic_clientset.go

```diff
@@ -64,6 +64,15 @@ func (cs *Clientset) Kind(apiVersion, kind, namespace string) (*ResourceClient,
 	return cs.resource(apiResource, namespace)
 }
 
+func getResourceGVR(apiResource *dynamicdiscovery.APIResource) schema.GroupVersionResource {
+	resource := apiResource.APIResource
+	return schema.GroupVersionResource{
+		Group:    resource.Group,
+		Version:  resource.Version,
+		Resource: resource.Name,
+	}
+}
+
 func (cs *Clientset) resource(apiResource *dynamicdiscovery.APIResource, namespace string) (*ResourceClient, error) {
 	// Create dynamic client for this apiVersion/resource.
 	gv := apiResource.GroupVersion()
```

Comment:
Can you check if the new dynamic client lets us leave GroupVersion unset? Ideally we should only need to call `dynamic.NewForConfig()` once in this package's `New()` function, and store that. I believe it was a limitation of the old dynamic client that we had to make a new one for different group-versions.

If you're able to remove the need to create a new client, you can also simplify this function by removing the `error` return.

Reply:
Yes, the new dynamic client allows `GroupVersion` to be unset, but it does check correctness of the format. Refer [this](https://github.com/kubernetes/apimachinery/blob/master/pkg/runtime/schema/group_version.go#L215-L231) for more information.

I will remove `NewForConfig` and `error` return.